src/stim/py/compiled_measurement_sampler_pybind_test.py

```diff
@@ -118,6 +119,37 @@ def test_skip_reference_sample():
         [[True]],
     )
 
+def test_reference_sample_init():
+    np.testing.assert_array_equal(
```

Comment:
Test a mix of commuting and anticommuting and redundant measurements, such as `MPP X1*X2 Y1*Y2 Z1*Z2`.

Note you can use `seed=0` to force the same underlying samples between a no-ref-sample and yes-ref-sample sampler to be identical, allowing you to check the ref is used correctly. Note seeds change meaning from version to version so you need to do the comparison, not just hardcode.